src/units/abilities.cpp

```diff
@@ -1168,52 +1168,112 @@ void attack_type::modified_attacks(unsigned & min_attacks,
 	}
 }
 
-//Functions used for change damage_type list with damage
-static std::optional<std::string> select_damage_type(const unit_ability_list& abil_list, const std::string& type)
+static std::string select_replacement_type(const unit_ability_list& damage_type_list)
 {
+	std::unordered_map<std::string, unsigned int> type_count;
+	unsigned int max = 0;
+	for(auto& i : damage_type_list) {
+		const config& c = *i.ability_cfg;
+		if(c.has_attribute("replacement_type")) {
+			std::string type = c["replacement_type"].str();
+			unsigned int count = ++type_count[type];
+			if((count > max)) {
+				max = count;
+			}
+		}
+	}
+
+	if (type_count.empty()) return "";
+
 	std::vector<std::string> type_list;
-	for(auto& i : abil_list) {
-		if(!(*i.ability_cfg)[type].str().empty()){
-			type_list.push_back((*i.ability_cfg)[type].str());
+	for(auto& i : type_count){
+		if(i.second == max){
+			type_list.push_back(i.first);
 		}
 	}
-	if(type_list.size() >= 2){
-		std::sort(type_list.begin(), type_list.end());
-		if(type_list.size() >= 3){
-			std::unordered_map<std::string, unsigned int> type_count;
-			for( const std::string& character : type_list ){
-				type_count[character]++;
+
+	if(type_list.empty()) return "";
+
+	return type_list.front();
```

Comment:
Either sort the list or use map instead of unordered_map for `type_count`.

Reply:
fixed by using map